src/python/foglamp/translators/omf_translator.py

```diff
@@ -0,0 +1,732 @@
+            pass
+
+
+
+        if data_available == True:
+            debug_msg_write("INFO", "OMF Message   |{0}| ".format(data_values_JSON))
+
+        else:
+            status = False
+            _log.warning(_message_list["e000009"])
+
+
+    except:
+        status  = False
+        message = _message_list["e000010"].format(e)
+
+        _log.error(message)
+        raise Exception(message)
+
+
+    return status, data_values_JSON
+
+
+def send_OMF_message_to_end_point(message_type, OMF_data):
+    """Sends data for OMF
+
+    Args:
+        message_type: possible values - Type | Container | Data
+        OMF_data:     message to send
+
+    Returns:
+        status:    True if successful, False otherwise.
+
+    Raises:
+        Exception: an error occurred during the OMF request
+
+    """
+
+    status = True
+
+    try:
+        msg_header = {'producertoken': _producer_token,
+                      'messagetype':   message_type,
+                      'action':        'create',
+                      'messageformat': 'JSON',
+                      'omfversion':    '1.0'}
+
+        response = requests.post(_relay_url, headers=msg_header, data=json.dumps(OMF_data), verify=False, timeout=30)
+
+        debug_msg_write("INFO", "Response |{0}| message: |{1}| |{2}| ".format(message_type,
+                                                                              response.status_code,
+                                                                              response.text))
+
+
+    except Exception as e:
+        status  = False
+        message = _message_list["e000007"].format(e)
+
+        _log.error(message)
+        raise Exception(message)
+
+    return status
+
+
+
+def log_configuration():
+    """Configures the log mechanism
+
+    Returns:
+        status:    True if successful, False otherwise.
+
+    Raises:
+        Exception: Fails to configure the log
+
+    """
+
+    global _log
+    global _handler
+    global _formatter
+
+    status = True
+
+    try:
+        _log = logging.getLogger(__module__)
+
+        _log.setLevel(logging.DEBUG)
+        _handler = logging.handlers.SysLogHandler(address='/dev/log')  # /var/run/syslog
+
+        _formatter = logging.Formatter('%(module)s.%(funcName)s: %(message)s')
+        _handler.setFormatter(_formatter)
+
+        _log.addHandler(_handler)
+
+    except Exception as e:
+        status = False
+        raise Exception(_message_list["e000005"].format(e))
+
+
+    return status
+
+
+def position_read():
+    """Retrieves the starting point for the send operation
+
+    Returns:
+        status:    True if successful, False otherwise.
+        position:
+
+    Raises:
+        Exception: operations at db level failed
+
+    Todo:
+        it should evolve using the DB layer
+    """
+
+    global log
+    global _pg_conn
+    global _pg_cur
+
+    status    = True
+    position  = 0
+
+    try:
+        sql_cmd = "SELECT reading_id FROM foglamp.streams WHERE id=1"
```

Comment:
I don;t see reading_id column in streams table in src/sql/foglamp_ddl.sql in `develop` branch; are we expecting it to be change before this PR is merged?